Core/src/test/java/com/rakuten/tech/mobile/perf/core/SenderThreadSpec.java

```diff
@@ -71,60 +70,40 @@ public void uncaughtException(Thread t, Throwable e) {
           fail("uncaught exception");
         }
       });
-      senderThread.start();
-      Thread.sleep(100);
-      senderThread.terminate();
-      loop(senderThread, 1);
+      startSenderThread(senderThread);
+
+      verify(sender, timeout(50).atLeast(1)).send(anyInt());
+
+      terminateSenderThread(senderThread);
 
       assertThat(count.get()).isEqualTo(0);
     }
 
-    @Test public void shouldBackOffAfterFailures() throws InterruptedException, IOException {
+    @Test public void shouldBackOffAfterFailures() throws IOException {
       doThrow(NullPointerException.class).when(sender).send(anyInt());
 
-      senderThread.start();
-      Thread.sleep(140);
-      senderThread.terminate();
-      loop(senderThread, 1); // give it some time to finish run()
+      startSenderThread(senderThread);
 
-      verify(sender, atMost(4)).send(anyInt());
+      verify(sender, timeout(100).times(4)).send(anyInt());
     }
 
-    @Test public void shouldResetBackOffAfterSuccess() throws InterruptedException,
-        IOException {
+    @Test public void shouldResetBackOffAfterSuccess() throws IOException {
       doThrow(NullPointerException.class).when(sender).send(anyInt());
 
-      senderThread.start();
-      Thread.sleep(140);
+      startSenderThread(senderThread);
 
+      // Failures to increase the back off timeout
+      verify(sender, timeout(100).atLeast(3)).send(anyInt());
       clearInvocations(sender);
-      doReturn(0).when(sender).send(anyInt());
-      loop(senderThread, 1); // give it some time to finish run()
-
-      Thread.sleep(50);
-
-      verify(sender, atLeast(4)).send(anyInt());
-    }
-
-
-    @Test public void shouldResetFailureCountAfterSuccess()
```

Comment:
I removed this test because it looked to me like it wasn't actually testing anything at all, unless I misunderstood something?

Also, `shouldResetFailureCountAfterSuccess ` sounds like it's testing exactly the same thing as the above test `shouldResetBackOffAfterSuccess`.